### PR TITLE
feat: expose tnx_id of cluster_rpc for prometheus

### PR DESCRIPTION
**Pull request by HJianBo**
Fixes https://emqx.atlassian.net/browse/EMQX-13323

Release version: v/e5.8.2

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- ~Added property-based tests for code which performs user input validation~
- ~Changed lines covered in coverage report~
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- ~Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket~
- ~Schema changes are backward compatible~

## Checklist for CI (.github/workflows) changes

- ~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- ~Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

File: apps/emqx_conf/src/emqx_cluster_rpc.erl [hunk 1]
```diff
@@ -486,6 +487,10 @@ get_cluster_tnx_id() ->
         Id -> Id
     end.
 
+get_current_tnx_id() ->
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_cluster_rpc.erl:490]: can use dirty read

File: apps/emqx_prometheus/src/emqx_prometheus.erl [hunk 2]
```diff
@@ -310,6 +313,7 @@ aggre_or_zip_init_acc() ->
         emqx_olp_data => meta_to_init_from(olp_metric_meta()),
         emqx_acl_data => meta_to_init_from(acl_metric_meta()),
         emqx_authn_data => meta_to_init_from(authn_metric_meta()),
+        cluster_rpc => meta_to_init_from(cluster_rpc_meta()),
```
Comment (thalesmg) [inline apps/emqx_prometheus/src/emqx_prometheus.erl:316]: It's a bit hard to follow the path for those prometheus callbacks, but it seems like putting this here might lead to the transaction id being summed over the cluster<sup>[[1]](https://github.com/emqx/emqx/blob/22e595fa96fdaf75fc9a02ae9e6146a1a1b31bb6/apps/emqx_prometheus/src/emqx_prometheus_cluster.erl#L81)</sup><sup>[[2]](https://github.com/emqx/emqx/blob/22e595fa96fdaf75fc9a02ae9e6146a1a1b31bb6/apps/emqx_prometheus/src/emqx_prometheus_cluster.erl#L116)</sup>?

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_prometheus/src/emqx_prometheus.erl [hunk 3]
```diff
@@ -495,6 +499,8 @@ emqx_collect(K = emqx_license_expiry_at, D) -> gauge_metric(?MG(K, D));
 %%--------------------------------------------------------------------
 %% Certs
 emqx_collect(K = emqx_cert_expiry_at, D) -> gauge_metrics(?MG(K, D));
+%% Cluster RPC
+emqx_collect(K = emqx_cluster_rpc_txid, D) -> gauge_metrics(?MG(K, D));
```
Comment (zmstone) [inline apps/emqx_prometheus/src/emqx_prometheus.erl:503]: cluster-rpc is an internal term, when presented to users, it's conf-sync.
```suggestion
emqx_collect(K = emqx_conf_sync_txid, D) -> gauge_metrics(?MG(K, D));
```

File: apps/emqx_prometheus/src/emqx_prometheus.erl (hunk 2 above)
Comment (HJianBo) [reply apps/emqx_prometheus/src/emqx_prometheus.erl:316]: Honestly, it doesn't add up.
This is because when it comes to getting aggregated metrics, its raw data carries the node as Label, e.g:
```
curl -X 'GET' \
  'http://127.0.0.1:18083/api/v5/prometheus/stats?mode=all_nodes_aggregated' \
  -H 'accept: text/plain' | grep emqx_conf

# TYPE emqx_conf_sync_txid gauge
# HELP emqx_conf_sync_txid
emqx_conf_sync_txid{node="emqx@127.0.0.1"} 6
```

for the node metrics, it looks like:
```
curl -X 'GET' \
  'http://127.0.0.1:18083/api/v5/prometheus/stats?mode=node' \
  -H 'accept: text/plain' | grep emqx_conf

# TYPE emqx_conf_sync_txid gauge
# HELP emqx_conf_sync_txid
emqx_conf_sync_txid 6
```

**Review by savonarola — APPROVED** [review]

**Review by qzhuyan — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-4872)
